### ISSUE TITLE
./waf build fail

### ISSUE THREAD

**Issue by gasper321** [MTG/essentia#237]
Hello. 

I'm following the tutorial for compiling, and when i get to ./waf part, i keep getting:

gasper@gasper-P5KR:~/CMRLJI/essentia$ ./waf 
Waf: Entering directory `/home/gasper/CMRLJI/essentia/build'
→ building from /home/gasper/CMRLJI/essentia
→ building from /home/gasper/CMRLJI/essentia/src
Building all the algorithms
Ignoring the following algorithms: GaiaTransform
→ building from /home/gasper/CMRLJI/essentia/src/examples
→ building from /home/gasper/CMRLJI/essentia/src/python
[ 23/297] cxx: src/essentia/utils/audiocontext.cpp -> build/src/essentia/utils/audiocontext.cpp.1.o
[ 58/297] cxx: src/algorithms/tonal/chordsdescriptors.cpp -> build/src/algorithms/tonal/chordsdescriptors.cpp.1.o
[ 59/297] cxx: src/algorithms/rhythm/onsetdetection.cpp -> build/src/algorithms/rhythm/onsetdetection.cpp.1.o
[ 60/297] cxx: src/algorithms/stats/variance.cpp -> build/src/algorithms/stats/variance.cpp.1.o
../src/essentia/utils/audiocontext.cpp: In member function ‘void essentia::AudioContext::encodePacket(int)’:
../src/essentia/utils/audiocontext.cpp:337:34: error: ‘_bufferFmt’ was not declared in this scope
                     (uint8_t**) &_bufferFmt, outputBufferSamples,
                                  ^
../src/essentia/utils/audiocontext.cpp:337:46: error: ‘outputBufferSamples’ was not declared in this scope
                     (uint8_t**) &_bufferFmt, outputBufferSamples,
                                              ^
../src/essentia/utils/audiocontext.cpp:338:52: error: expected ‘;’ before ‘)’ token
                     (const uint8_t**) _buffer, size) < 0) {
                                                    ^
../src/essentia/utils/audiocontext.cpp:301:7: warning: unused variable ‘inputPlaneSize’ [-Wunused-variable]
   int inputPlaneSize = av_samples_get_buffer_size(NULL, 
       ^
Waf: Leaving directory `/home/gasper/CMRLJI/essentia/build'
Build failed
 -> task in 'essentia' failed (exit status 1): 
	{task 139874326255376: cxx audiocontext.cpp -> audiocontext.cpp.1.o}
['/usr/bin/g++', '-pipe', '-Wall', '-msse', '-msse2', '-mfpmath=sse', '-O2', '-fPIC', '-pthread', '-pthread', '-pthread', '-I/home/gasper/CMRLJI/essentia/build/src', '-I/home/gasper/CMRLJI/essentia/src', '-I/home/gasper/CMRLJI/essentia/build/src/essentia', '-I/home/gasper/CMRLJI/essentia/src/essentia', '-I/home/gasper/CMRLJI/essentia/build/src/essentia/scheduler', '-I/home/gasper/CMRLJI/essentia/src/essentia/scheduler', '-I/home/gasper/CMRLJI/essentia/build/src/essentia/streaming', '-I/home/gasper/CMRLJI/essentia/src/essentia/streaming', '-I/home/gasper/CMRLJI/essentia/build/src/essentia/streaming/algorithms', '-I/home/gasper/CMRLJI/essentia/src/essentia/streaming/algorithms', '-I/home/gasper/CMRLJI/essentia/build/src/essentia/utils', '-I/home/gasper/CMRLJI/essentia/src/essentia/utils', '-I/home/gasper/CMRLJI/essentia/build/src/3rdparty', '-I/home/gasper/CMRLJI/essentia/src/3rdparty', '-I/usr/local/include', '-I/usr/include/taglib', '-DHAVE_AVCODEC=1', '-DHAVE_AVFORMAT=1', '-DHAVE_AVUTIL=1', '-DHAVE_SWRESAMPLE=1', '-DHAVE_SAMPLERATE=1', '-DHAVE_TAGLIB=1', '-DHAVE_YAML=1', '-DHAVE_FFTW=1', '-D__STDC_CONSTANT_MACROS', '-DPYTHONDIR="/usr/local/lib/python2.7/dist-packages"', '-DPYTHONARCHDIR="/usr/local/lib/python2.7/dist-packages"', '-DHAVE_PYTHON_H=1', '../src/essentia/utils/audiocontext.cpp', '-c', '-o', 'src/essentia/utils/audiocontext.cpp.1.o']

The interesting thing is, that if i keep rerunning ./waf, i keep getting 3 steps further...like 54-57 ERROR, 57-60 ERROR, etc.

Any help much appreciated.

Oh btw i ran sudo ./waf configure --mode=release --with-python --with-examples
to configure first, and everything seemed okay.

Regards, Gasper

**Comment by julj** [comment]
Same here.

**Comment by julj** [comment]
Similar issue here.

```
[ 24/282] cxx: src/essentia/utils/splineutil.cpp -> build/src/essentia/utils/splineutil.cpp.1.o
In file included from ../src/essentia/utils/ffmpegapi.h:100:0,
                 from ../src/essentia/utils/audiocontext.h:26,
                 from ../src/essentia/utils/audiocontext.cpp:20:
/home/dude/tmp/essentia-2.0.1/src/3rdparty/audioconvert.h:55:6: error: use of enum ‘SampleFormat’ without previous declaration
 enum SampleFormat avcodec_get_sample_fmt(const char* name);
      ^
/home/dude/tmp/essentia-2.0.1/src/3rdparty/audioconvert.h:55:59: error: invalid type in declaration before ‘;’ token
 enum SampleFormat avcodec_get_sample_fmt(const char* name);
                                                           ^
/home/dude/tmp/essentia-2.0.1/src/3rdparty/audioconvert.h:94:45: error: use of enum ‘SampleFormat’ without previous declaration
 AVAudioConvert *av_audio_convert_alloc(enum SampleFormat out_fmt, int out_channels,
                                             ^
/home/dude/tmp/essentia-2.0.1/src/3rdparty/audioconvert.h:95:45: error: use of enum ‘SampleFormat’ without previous declaration
                                        enum SampleFormat in_fmt, int in_channels,
                                             ^
../src/essentia/utils/audiocontext.cpp: In member function ‘void essentia::AudioContext::encodePacket(int)’:
../src/essentia/utils/audiocontext.cpp:278:17: warning: ‘int avcodec_encode_audio(AVCodecContext*, uint8_t*, int, const short int*)’ is deprecated (declared at /usr/include/libavcodec/avcodec.h:4355) [-Wdeprecated-declarations]
   packet.size = avcodec_encode_audio(_codecCtx, _outputBuffer, frame_bytes, (short*)_inputBuffer);
                 ^
../src/essentia/utils/audiocontext.cpp:278:97: warning: ‘int avcodec_encode_audio(AVCodecContext*, uint8_t*, int, const short int*)’ is deprecated (declared at /usr/include/libavcodec/avcodec.h:4355) [-Wdeprecated-declarations]
   packet.size = avcodec_encode_audio(_codecCtx, _outputBuffer, frame_bytes, (short*)_inputBuffer);
                                                                                                 ^
../src/essentia/utils/audiocontext.cpp: In member function ‘void essentia::AudioContext::writeEOF()’:
../src/essentia/utils/audiocontext.cpp:323:14: warning: ‘int avcodec_encode_audio(AVCodecContext*, uint8_t*, int, const short int*)’ is deprecated (declared at /usr/include/libavcodec/avcodec.h:4355) [-Wdeprecated-declarations]
       size = avcodec_encode_audio(_codecCtx, _outputBuffer, frame_bytes, (short*)_inputBuffer); // outputs remaining samples
              ^
../src/essentia/utils/audiocontext.cpp:323:94: warning: ‘int avcodec_encode_audio(AVCodecContext*, uint8_t*, int, const short int*)’ is deprecated (declared at /usr/include/libavcodec/avcodec.h:4355) [-Wdeprecated-declarations]
       size = avcodec_encode_audio(_codecCtx, _outputBuffer, frame_bytes, (short*)_inputBuffer); // outputs remaining samples
                                                                                              ^
../src/essentia/utils/audiocontext.cpp:327:14: warning: ‘int avcodec_encode_audio(AVCodecContext*, uint8_t*, int, const short int*)’ is deprecated (declared at /usr/include/libavcodec/avcodec.h:4355) [-Wdeprecated-declarations]
       size = avcodec_encode_audio(_codecCtx, _outputBuffer, frame_bytes, NULL); // sets EOF
              ^
../src/essentia/utils/audiocontext.cpp:327:78: warning: ‘int avcodec_encode_audio(AVCodecContext*, uint8_t*, int, const short int*)’ is deprecated (declared at /usr/include/libavcodec/avcodec.h:4355) [-Wdeprecated-declarations]
       size = avcodec_encode_audio(_codecCtx, _outputBuffer, frame_bytes, NULL); // sets EOF
                                                                              ^
Waf: Leaving directory `/home/dude/tmp/essentia-2.0.1/build'
Build failed
 -> task in 'essentia' failed (exit status 1): 
	{task 140199393069520: cxx audiocontext.cpp -> audiocontext.cpp.1.o}
['/usr/bin/g++', '-pipe', '-Wall', '-O2', '-fPIC', '-I/home/dude/tmp/essentia-2.0.1/build/src', '-I/home/dude/tmp/essentia-2.0.1/src', '-I/home/dude/tmp/essentia-2.0.1/build/src/essentia', '-I/home/dude/tmp/essentia-2.0.1/src/essentia', '-I/home/dude/tmp/essentia-2.0.1/build/src/essentia/scheduler', '-I/home/dude/tmp/essentia-2.0.1/src/essentia/scheduler', '-I/home/dude/tmp/essentia-2.0.1/build/src/essentia/streaming', '-I/home/dude/tmp/essentia-2.0.1/src/essentia/streaming', '-I/home/dude/tmp/essentia-2.0.1/build/src/essentia/streaming/algorithms', '-I/home/dude/tmp/essentia-2.0.1/src/essentia/streaming/algorithms', '-I/home/dude/tmp/essentia-2.0.1/build/src/essentia/utils', '-I/home/dude/tmp/essentia-2.0.1/src/essentia/utils', '-I/home/dude/tmp/essentia-2.0.1/build/src/3rdparty', '-I/home/dude/tmp/essentia-2.0.1/src/3rdparty', '-I/usr/include/taglib', '-DHAVE_AVCODEC=1', '-DHAVE_AVFORMAT=1', '-DHAVE_AVUTIL=1', '-DHAVE_TAGLIB=1', '-D__STDC_CONSTANT_MACROS', '-DHAVE_YAML=1', '-DHAVE_FFTW=1', '-DHAVE_SAMPLERATE=1', '-DPYTHONDIR="/usr/local/lib/python2.7/dist-packages"', '-DPYTHONARCHDIR="/usr/local/lib/python2.7/dist-packages"', '-DHAVE_PYTHON_H=1', '../src/essentia/utils/audiocontext.cpp', '-c', '-o', 'src/essentia/utils/audiocontext.cpp.1.o']
```

**Comment by gasper321** [comment]
I tried reinstalling the dependencies and rerunning config and then compiling.

The original error is the same as Julien pasted and it always first happens at step 24.

[ 24/304] cxx: src/essentia/utils/jsonconvert.cpp -> build/src/essentia/utils/jsonconvert.cpp.1.o
../src/essentia/utils/audiocontext.cpp: In member function ‘void essentia::AudioContext::encodePacket(int)’:
../src/essentia/utils/audiocontext.cpp:337:34: error: ‘_bufferFmt’ was not declared in this scope
                     (uint8_t**) &_bufferFmt, outputBufferSamples,
                                  ^
../src/essentia/utils/audiocontext.cpp:337:46: error: ‘outputBufferSamples’ was not declared in this scope
                     (uint8_t**) &_bufferFmt, outputBufferSamples,
                                              ^
../src/essentia/utils/audiocontext.cpp:338:52: error: expected ‘;’ before ‘)’ token
                     (const uint8_t**) _buffer, size) < 0) {
                                                    ^
../src/essentia/utils/audiocontext.cpp:301:7: warning: unused variable ‘inputPlaneSize’ [-Wunused-variable]
   int inputPlaneSize = av_samples_get_buffer_size(NULL, 
       ^
Waf: Leaving directory `/home/gasper/CMRLJI/essentia/build'
Build failed
 -> task in 'essentia' failed (exit status 1): 
	{task 139915696203280: cxx audiocontext.cpp -> audiocontext.cpp.1.o}
['/usr/bin/g++', '-pipe', '-Wall', '-msse', '-msse2', '-mfpmath=sse', '-O2', '-fPIC', '-pthread', '-pthread', '-pthread', '-I/home/gasper/CMRLJI/essentia/build/src', '-I/home/gasper/CMRLJI/essentia/src', '-I/home/gasper/CMRLJI/essentia/build/src/essentia', '-I/home/gasper/CMRLJI/essentia/src/essentia', '-I/home/gasper/CMRLJI/essentia/build/src/essentia/scheduler', '-I/home/gasper/CMRLJI/essentia/src/essentia/scheduler', '-I/home/gasper/CMRLJI/essentia/build/src/essentia/streaming', '-I/home/gasper/CMRLJI/essentia/src/essentia/streaming', '-I/home/gasper/CMRLJI/essentia/build/src/essentia/streaming/algorithms', '-I/home/gasper/CMRLJI/essentia/src/essentia/streaming/algorithms', '-I/home/gasper/CMRLJI/essentia/build/src/essentia/utils', '-I/home/gasper/CMRLJI/essentia/src/essentia/utils', '-I/home/gasper/CMRLJI/essentia/build/src/3rdparty', '-I/home/gasper/CMRLJI/essentia/src/3rdparty', '-I/home/gasper/CMRLJI/essentia/build/src/3rdparty/vamp-plugin-sdk-2.4', '-I/home/gasper/CMRLJI/essentia/src/3rdparty/vamp-plugin-sdk-2.4', '-I/usr/local/include', '-I/usr/include/taglib', '-DHAVE_AVCODEC=1', '-DHAVE_AVFORMAT=1', '-DHAVE_AVUTIL=1', '-DHAVE_SWRESAMPLE=1', '-DHAVE_SAMPLERATE=1', '-DHAVE_TAGLIB=1', '-DHAVE_YAML=1', '-DHAVE_FFTW=1', '-D__STDC_CONSTANT_MACROS', '-DPYTHONDIR="/usr/local/lib/python2.7/dist-packages"', '-DPYTHONARCHDIR="/usr/local/lib/python2.7/dist-packages"', '-DHAVE_PYTHON_H=1', '../src/essentia/utils/audiocontext.cpp', '-c', '-o', 'src/essentia/utils/audiocontext.cpp.1.o']

**Comment by julj** [comment]
The issue seems to be with libavresample:

```
Checking for 'libavresample'             : not found 
Checking for 'libswresample'             : not found 
```

though it is installed (/usr/lib/x86_64-linux-gnu/libavresample.so.1.1.0)

**Comment by gasper321** [comment]
For me only libavresample is not found, even though packages:
Libavresample-dev 6:9.18
Libavresample1 6:9.18

are installed. Ffmpeg latest version was downloaded and compiled aswell and
is detected.

Any ideas?

**Comment by dbogdanov** [comment]
Check if pkg-config is able to find libavresample, you can see the log in ```build/config.log```

**Comment by dbogdanov** [comment]
@julj what is your version of libav? it seems to be too new for Essentia 2.0

2.0.1 works with the libav version 6:9.18 from ubuntu trusty (10.04) repository

2.1 works with the libav version 6:11-1 from ubuntu utopic (10.10) repository, and you need to install libavresample-dev

**Comment by julj** [comment]
I managed to get Essentia 2.0.1 working with latest libav* libraries in Ubuntu 14.04. I just had to add libavresample-dev.

**Comment by gasper321** [comment]
Hi.

@julj how did you install ffmpeg (libavcodec)? did you download it and build it from source? Because now i'm getting a new error:

[278/289] cxxshlib: build/src/examples/vampplugins.cpp.23.o build/src/examples/vamppluginsextra.cpp.23.o build/src/examples/vampwrapper.cpp.23.o build/src/3rdparty/vamp-plugin-sdk-2.4/src/vamp-sdk/FFT.cpp.23.o build/src/3rdparty/vamp-plugin-sdk-2.4/src/vamp-sdk/PluginAdapter.cpp.23.o build/src/3rdparty/vamp-plugin-sdk-2.4/src/vamp-sdk/RealTime.cpp.23.o -> build/src/examples/libvamp_essentia.so
/usr/bin/ld: /usr/local/lib/libavformat.a(allformats.o): relocation R_X86_64_32 against `ff_a64_muxer' can not be used when making a shared object; recompile with -fPIC
/usr/local/lib/libavformat.a: error adding symbols: Bad value
collect2: error: ld returned 1 exit status
Waf: Leaving directory `/home/joze/essentia-2.0.1/build'
Build failed
 -> task in 'vamp_essentia' failed (exit status 1): 
	{task 140399579854288: cxxshlib vampplugins.cpp.23.o,vamppluginsextra.cpp.23.o,vampwrapper.cpp.23.o,FFT.cpp.23.o,PluginAdapter.cpp.23.o,RealTime.cpp.23.o -> libvamp_essentia.so}
['/usr/bin/g++', '-shared', '-pthread', '-pthread', 'src/examples/vampplugins.cpp.23.o', 'src/examples/vamppluginsextra.cpp.23.o', 'src/examples/vampwrapper.cpp.23.o', 'src/3rdparty/vamp-plugin-sdk-2.4/src/vamp-sdk/FFT.cpp.23.o', 'src/3rdparty/vamp-plugin-sdk-2.4/src/vamp-sdk/PluginAdapter.cpp.23.o', 'src/3rdparty/vamp-plugin-sdk-2.4/src/vamp-sdk/RealTime.cpp.23.o', '-o', '/home/joze/essentia-2.0.1/build/src/examples/libvamp_essentia.so', '-Wl,-Bstatic', '-Lsrc', '-lessentia', '-Wl,-Bdynamic', '-L/usr/local/lib', '-L/usr/local/lib', '-L/usr/local/lib', '-L/usr/local/lib', '-lfftw3f', '-lyaml', '-lavformat', '-lavcodec', '-lz', '-lswresample', '-lavutil', '-lm', '-lavcodec', '-lz', '-lswresample', '-lavutil', '-lm', '-lavutil', '-lm', '-lswresample', '-lavutil', '-lm', '-lsamplerate', '-ltag']

**I'm pretty sure it has something to do with versions of dependencies, but im not exactly sure. Right now i have essentia 2.0.1 and config seems okay:**

joze@joze-desktop:~/essentia-2.0.1$ ./waf configure --mode=release --with-python --with-examples --with-vamp 
Setting top to                           : /home/joze/essentia-2.0.1 
Setting out to                           : /home/joze/essentia-2.0.1/build 
→ configuring the project in /home/joze/essentia-2.0.1
→ Building in release mode
Checking for 'g++' (c++ compiler)        : /usr/bin/g++ 
Checking for 'gcc' (c compiler)          : /usr/bin/gcc 
Checking for program pkg-config          : /usr/bin/pkg-config 
Checking for 'libavcodec'                : yes 
Checking for 'libavformat'               : yes 
Checking for 'libavutil'                 : yes 
Checking for 'libswresample'             : yes 
Checking for 'taglib'                    : yes 
Checking for 'yaml-0.1'                  : yes 
Checking for 'fftw3f'                    : yes 
Checking for 'samplerate'                : yes 
Checking for 'gaia2'                     : not found 
Checking for program python              : /usr/bin/python 
Checking for python version              : (2, 7, 6, 'final', 0) 
Checking for library python2.7 in LIBDIR : yes 
Checking for program /usr/bin/python-config,python2.7-config,python-config-2.7,python2.7m-config : /usr/bin/python-config 
Checking for header Python.h                                                                     : yes 

================================ CONFIGURATION SUMMARY ================================
- FFmpeg / libav detected!
  The following algorithms will be included: ['AudioLoader', 'MonoLoader', 'EqloudLoader', 'EasyLoader', 'MonoWriter', 'AudioWriter']

- libsamplerate (SRC) detected!
  The following algorithms will be included: ['Resample']

- TagLib detected!
  The following algorithms will be included: ['MetadataReader']

- Gaia2 seems to be missing.
  The following algorithms will be ignored: ['GaiaTransform']

=======================================================================================
'configure' finished successfully (0.419s)

Here are the versions of the dependencies:
i  libavcodec-dev                                6:9.18-0ubuntu0.14.04.1                                       amd64        Development files for libavcodec
ii  libavcodec54:amd64                            6:9.18-0ubuntu0.14.04.1                                       amd64        Libav codec library
ii  libavformat-dev                               6:9.18-0ubuntu0.14.04.1                                       amd64        Development files for libavformat
ii  libavformat54:amd64                           6:9.18-0ubuntu0.14.04.1                                       amd64        Libav file format library

ffmpeg version 2.6.1 Copyright (c) 2000-2015 the FFmpeg developers
  built with gcc 4.8 (Ubuntu 4.8.2-19ubuntu1)
  configuration: 
  libavutil      54. 20.100 / 54. 20.100
  libavcodec     56. 26.100 / 56. 26.100
  libavformat    56. 25.101 / 56. 25.101
  libavdevice    56.  4.100 / 56.  4.100
  libavfilter     5. 11.102 /  5. 11.102
  libswscale      3.  1.101 /  3.  1.101
  libswresample   1.  1.100 /  1.  1.100

Any help appreciated. Thank you.

**Comment by julj** [comment]
Sorry, forget my previous comment. I uninstalled libavresample-dev and used libswresample instead, and it worked (with some warnings though). Here's all the libav* and libsw* libraries I have (not filtered)

```
dude@here:~$ dpkg -l | grep libav
ii  gstreamer1.0-libav:amd64                              1.2.4-1~ubuntu1                                     amd64        libav plugin for GStreamer
rc  libav-tools                                           7:1.2.6-1~trusty1                                   all          Multimedia player, server, encoder and transcoder (transitional package)
ii  libavahi-client-dev                                   0.6.31-4ubuntu1                                     amd64        Development files for the Avahi client library
ii  libavahi-client3:amd64                                0.6.31-4ubuntu1                                     amd64        Avahi client library
ii  libavahi-client3:i386                                 0.6.31-4ubuntu1                                     i386         Avahi client library
ii  libavahi-common-data:amd64                            0.6.31-4ubuntu1                                     amd64        Avahi common data files
ii  libavahi-common-data:i386                             0.6.31-4ubuntu1                                     i386         Avahi common data files
ii  libavahi-common-dev                                   0.6.31-4ubuntu1                                     amd64        Development files for the Avahi common library
ii  libavahi-common3:amd64                                0.6.31-4ubuntu1                                     amd64        Avahi common library
ii  libavahi-common3:i386                                 0.6.31-4ubuntu1                                     i386         Avahi common library
ii  libavahi-core7:amd64                                  0.6.31-4ubuntu1                                     amd64        Avahi's embeddable mDNS/DNS-SD library
ii  libavahi-glib1:amd64                                  0.6.31-4ubuntu1                                     amd64        Avahi GLib integration library
ii  libavahi-gobject0:amd64                               0.6.31-4ubuntu1                                     amd64        Avahi GObject library
ii  libavbin-dev                                          7-1.4build1                                         amd64        development files for libavbin
ii  libavbin0                                             7-1.4build1                                         amd64        cross-platform media decoding library
ii  libavc1394-0:amd64                                    0.5.4-2                                             amd64        control IEEE 1394 audio/video devices
ii  libavcodec-dev                                        7:1.2.6-1~trusty1                                   amd64        Development files for libavcodec
rc  libavcodec-extra-54:amd64                             7:1.2.6-1~trusty1                                   amd64        FFmpeg codec library (additional codecs)
ii  libavcodec54:amd64                                    7:1.2.6-1~trusty1                                   amd64        FFmpeg codec library
rc  libavdevice53:amd64                                   7:1.2.6-1~trusty1                                   amd64        FFmpeg device handling library
rc  libavfilter3:amd64                                    7:1.2.6-1~trusty1                                   amd64        FFmpeg video filtering library
ii  libavformat-dev                                       7:1.2.6-1~trusty1                                   amd64        Development files for libavformat
ii  libavformat54:amd64                                   7:1.2.6-1~trusty1                                   amd64        FFmpeg file format library
ii  libavresample1:amd64                                  7:1.2.6-1~trusty1                                   amd64        FFmpeg audo resampling library
ii  libavutil-dev                                         7:1.2.6-1~trusty1                                   amd64        Development files for libavutil
ii  libavutil52:amd64                                     7:1.2.6-1~trusty1                                   amd64        FFmpeg utility library
dude@here:~$ dpkg -l | grep libsw
ii  libswresample-dev                                     7:1.2.6-1~trusty1                                   amd64        Development files for libswresample
ii  libswresample0:amd64                                  7:1.2.6-1~trusty1                                   amd64        FFmpeg video software resampling library
ii  libswscale2:amd64   
```

**Comment by gasper321** [comment]
@julj How did you install libswresample only? It is part of the ffmpeg package. Did you install ffmpeg from source? I'd like to know exactly what you did with libavresample and libswresample. Because if i search: 
dpkg -l | grep libsw (ffmpeg installed from ppa https://launchpad.net/~mc3man/+archive/ubuntu/trusty-media)

I dont get any results - it doesn't find anything.

Thanks for your help.

**Comment by julj** [comment]
I got it from Jon Severinsson's PPA: https://launchpad.net/~jon-severinsson/+archive/ubuntu/ffmpeg.

**Comment by gasper321** [comment]
This was resolved by using the appropriate version of ffmpeg, as linked by julj in combination with essentia version 2.0.1. The instructions for installation were since updated to better reflect the required versions. Thanks to developer.

Thank you everyone for your help.

**Comment by yyf** [comment]
Similar issue here trying to ./waf Essentia-2.0.1 on OSX 10.8.5, any idea how to proceed? Thanks. 

Build failed
 -> task in 'essentia' failed (exit status 1): 
	{task 4376613136: cxx ringbufferoutput.cpp -> ringbufferoutput.cpp.1.o}
[	'clang++', 
	'-stdlib=libstdc++', 
	'-std=c++11', 
	'-Wno-gnu', 
	'-Wno-static-float-init', 
	'-I/usr/local/include', 
	'-I/Users/.../.../.../essentia-2.0.1/build/src', 
	'-I/Users/.../.../.../essentia-2.0.1/src', 
	'-I/Users/.../.../.../essentia-2.0.1/build/src/essentia', 
	'-I/Users/.../.../.../essentia-2.0.1/src/essentia', 
	'-I/Users/.../.../.../essentia-2.0.1/build/src/essentia/scheduler', 
	'-I/Users/.../.../.../essentia-2.0.1/src/essentia/scheduler', 
	'-I/Users/.../.../.../essentia-2.0.1/build/src/essentia/streaming', 
	'-I/Users/.../.../.../essentia-2.0.1/src/essentia/streaming', 
	'-I/Users/.../.../.../essentia-2.0.1/build/src/essentia/streaming/algorithms', 
	'-I/Users/.../.../.../essentia-2.0.1/src/essentia/streaming/algorithms', 
	'-I/Users/.../.../.../essentia-2.0.1/build/src/essentia/utils', 
	'-I/Users/.../.../.../essentia-2.0.1/src/essentia/utils', 
	'-I/Users/.../.../.../essentia-2.0.1/build/src/3rdparty', 
	'-I/Users/.../.../.../essentia-2.0.1/src/3rdparty', 
	'-I/Users/.../.../.../essentia-2.0.1/build/src/3rdparty/vamp-plugin-sdk-2.4', 
	'-I/Users/.../.../.../essentia-2.0.1/src/3rdparty/vamp-plugin-sdk-2.4', 
	'-I/opt/local/include', '-I/opt/local/include/taglib', 
	'-DGTEST_HAS_TR1_TUPLE=0', 
	'-DHAVE_AVCODEC=1', 
	'-DHAVE_AVFORMAT=1', 
	'-DHAVE_AVUTIL=1', 
	'-DHAVE_SWRESAMPLE=1', 
	'-DHAVE_TAGLIB=1', 
	'-D__STDC_CONSTANT_MACROS', 
	'-DHAVE_YAML=1', 
	'-DHAVE_FFTW=1', 
	'-DHAVE_SAMPLERATE=1', 
	'../src/algorithms/io/ringbufferoutput.cpp', 
	'-c', 
	'-o', 
	'src/algorithms/io/ringbufferoutput.cpp.1.o'
	]

**Comment by dbogdanov** [comment]
Compilation for 2.0/2.0.1 Essentia release on Ubuntu 14.04 is fixed in the corresponding bugfix branches. No need to go for workaround using libswresample anymore. 

@yyf We have fixes some issues with OSX in [2.1 beta2 release](https://github.com/MTG/essentia/releases/tag/v2.1_beta2)
